lib/fog/aws/requests/rds/modify_db_instance.rb

```diff
@@ -61,6 +61,9 @@ def modify_db_instance(db_name, apply_immediately, options={})
               #else
               #  modified_server = server["PendingModifiedValues"].merge!(options) # it appends
               #end
+              if options["NewDBInstanceIdentifier"]
+                options["DBInstanceIdentifier"] = options.delete("NewDBInstanceIdentifier")
```

Comment:
definitely broken parser.  seems that anything in `PendingModifiedValues` gets assigned to the model by the parser.